integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -156,4 +161,206 @@ object DataLoadProcessBuilderOnSpark {
+      DataLoadProcessorStepOnSpark.sortAndWriteFunc(rows, context.partitionId, modelBroadcast,
+        writeStepRowCounter, conf.value.value))
+
+    // Log the number of rows in each step
+    LOGGER.info("Total rows processed in step Input Processor: " + inputStepRowCounter.value)
+    LOGGER.info("Total rows processed in step Data Converter: " + convertStepRowCounter.value)
+    LOGGER.info("Total rows processed in step Sort Processor: " + sortStepRowCounter.value)
+    LOGGER.info("Total rows processed in step Data Writer: " + writeStepRowCounter.value)
+
+    // Update status
+    if (partialSuccessAccum.value != 0) {
+      val uniqueLoadStatusId = model.getTableName + CarbonCommonConstants.UNDERSCORE +
+                               "Partial_Success"
+      val loadMetadataDetails = new LoadMetadataDetails()
+      loadMetadataDetails.setSegmentStatus(SegmentStatus.LOAD_PARTIAL_SUCCESS)
+      val executionErrors = new ExecutionErrors(FailureCauses.NONE, "")
+      executionErrors.failureCauses = FailureCauses.BAD_RECORDS
+      Array((uniqueLoadStatusId, (loadMetadataDetails, executionErrors)))
+    } else {
+      val uniqueLoadStatusId = model.getTableName + CarbonCommonConstants.UNDERSCORE + "Success"
+      val loadMetadataDetails = new LoadMetadataDetails()
+      loadMetadataDetails.setSegmentStatus(SegmentStatus.SUCCESS)
+      val executionErrors = new ExecutionErrors(FailureCauses.NONE, "")
+      Array((uniqueLoadStatusId, (loadMetadataDetails, executionErrors)))
+    }
+  }
+
+  /**
+   * provide RDD for sample
+   * CSVRecordReader(univocity parser) will output only one column
+   */
+  private def getSampleRDD(
+      sparkSession: SparkSession,
+      model: CarbonLoadModel,
+      hadoopConf: Configuration,
+      configuration: CarbonDataLoadConfiguration,
+      modelBroadcast: Broadcast[CarbonLoadModel]
+  ): RDD[(Object, Object)] = {
+    // initialize and prepare row counter
+    val configuration = DataLoadProcessBuilder.createConfiguration(model)
+    val header = configuration.getHeader
+    val rangeColumn = model.getRangePartitionColumn
+    val rangeColumnIndex = (0 until header.length).find{
+      index =>
+        header(index).equalsIgnoreCase(rangeColumn.getColName)
+    }.get
+    val rangeField = configuration
+      .getDataFields
+      .find(dataField => dataField.getColumn.getColName.equals(rangeColumn.getColName))
+      .get
+
+    // 1. Input
+    val newHadoopConf = new Configuration(hadoopConf)
+    newHadoopConf
+      .set(CSVInputFormat.SELECT_COLUMN_INDEX, "" + rangeColumnIndex)
+    val inputRDD = CsvRDDHelper
+      .csvFileScanRDD(sparkSession, model, newHadoopConf)
+      .mapPartitionsWithIndex { case (index, rows) =>
+        DataLoadProcessorStepOnSpark
+          .internalSampleInputFunc(rows, rangeField, index, modelBroadcast)
+      }
+
+    // 2. Convert
+    val conf = SparkSQLUtil.broadCastHadoopConf(sparkSession.sparkContext, hadoopConf)
```

Comment:
Why don't you use the already brodcasted conf which is broadcasted in caller method.

Reply:
SerializableConfiguration is private in spark package